stl/inc/chrono

```diff
@@ -2309,15 +2337,109 @@ namespace chrono {
     }
 #endif // __cpp_lib_concepts
 
+    // [time.zone.link]
+
+    class time_zone_link {
+    public:
+        explicit time_zone_link(string_view _Name_, string_view _Target_) : _Name(_Name_), _Target(_Target_) {}
+
+        time_zone_link(time_zone_link&&) = default;
+        time_zone_link& operator=(time_zone_link&&) = default;
+
+        _NODISCARD string_view name() const noexcept {
+            return _Name;
+        }
+
+        _NODISCARD string_view target() const noexcept {
+            return _Target;
+        }
+
+    private:
+        string _Name;
+        string _Target;
+    };
+
+    _NODISCARD inline bool operator==(const time_zone_link& _Left, const time_zone_link& _Right) noexcept {
+        return _Left.name() == _Right.name();
+    }
+
+#ifdef __cpp_lib_concepts
+    _NODISCARD inline strong_ordering operator<=>(const time_zone_link& _Left, const time_zone_link& _Right) noexcept {
+        return _Left.name() <=> _Right.name();
+    }
+#endif // __cpp_lib_concepts
+
     // [time.zone.db]
 
-    // TRANSITION: work in progress
-    // STRUCT tzdb
+    _NODISCARD inline string _Xtzdb_generate_current_zone() {
+        unique_ptr<__std_tzdb_current_zone_info, decltype(&__std_tzdb_delete_current_zone)> _Info{
+            __std_tzdb_get_current_zone(), &__std_tzdb_delete_current_zone};
+        if (_Info == nullptr) {
+            _Xbad_alloc();
+        } else if (_Info->_Err == __std_tzdb_error::_Win_error) {
+            _XGetLastError();
+        } else if (_Info->_Err == __std_tzdb_error::_Icu_error) {
+            _Xruntime_error("Internal error loading IANA database information");
+        }
+
+        return {_Info->_Tz_name};
+    }
+
+    template <class _Ty>
+    _NODISCARD const _Ty* _Locate_zone_impl(const vector<_Ty>& _Vec, string_view _Name) {
+        const auto _Result = _STD find_if(_Vec.begin(), _Vec.end(), [&](auto& _Tz) { return _Tz.name() == _Name; });
+        return _Result != _Vec.end() ? &*_Result : nullptr;
+    }
+
     struct tzdb {
-        vector<leap_second, _Crt_allocator<leap_second>> leap_seconds;
+        vector<time_zone> time_zones;
+        vector<time_zone_link> links;
+        vector<leap_second> leap_seconds;
         bool _All_ls_positive;
+
+        _NODISCARD const time_zone* locate_zone(string_view _Tz_name) const {
+            auto _Tz = _Locate_zone_impl(time_zones, _Tz_name);
+            if (_Tz == nullptr) {
+                const auto _Link = _Locate_zone_impl(links, _Tz_name);
+                if (_Link != nullptr) {
+                    _Tz = _Locate_zone_impl(time_zones, _Link->target());
+                }
+            }
+
+            return _Tz;
+        }

```

Comment:
If the given `_Tz_name` isn't found in either `zones` or `links`, [we are supposed to throw a `runtime_error`](http://eel.is/c++draft/time.zone.db#tzdb-3) -- at this point, I think we just return `nullptr` if we don't find a matching zone/link.

Reply:
I was wondering why it was a pointer not a reference but it looks like that matters for zoned_time traits.